tools/codegen/utils.py

```diff
@@ -194,12 +194,15 @@ def write_sharded(
                 else:
                     shard[key] = []
 
-
         def merge_env(into: Dict[str, List[str]], from_: Dict[str, List[str]]) -> None:
             for k, v in from_.items():
                 assert k in sharded_keys, f"undeclared sharded key {k}"
                 into[k] += v
 
+        if self.dry_run:
+            # Dry runs don't write any templates, so incomplete environments are fine
+            items = ()
```

Comment:
Should we skip generating this altogether?

Reply:
We still need to call `write_with_template` to update the `filenames` list.

Reply:
Ho what I meant is that the caller doesn't need to provide a valid items and so could skip generating it?
I know the loading of the yaml is especially expensive, so if we can skip that, it would be a nice speedup.
But this is quite fast anyways so maybe not worth it?

Reply:
Oh I see. Once the `--per-operator-headers` flag lands then we really do need to read `native_functions.yaml` even for dry runs.

Reply:
Ok!
I'm working on the import of that other PR.
Let's delay this one until then then?

Reply:
Sure you can delay this, it's a fairly small gain in the grand scheme of things. To be clear though, this PR doesn't depend on the other stack.